ignite/pkg/cosmosclient/cosmosclient.go

```diff
@@ -571,7 +574,7 @@ func (c Client) CreateTx(goCtx context.Context, account cosmosaccount.Account, m
 			return TxService{}, errors.WithStack(err)
 		}
 		// the simulated gas can vary from the actual gas needed for a real transaction
-		// we add an additional amount to ensure sufficient gas is provided
+		// we add an amount to ensure sufficient gas is provided
 		gas += 20000
```

Comment:
If I'm not wrong it's me who introduced this. While testing it appeared that the value returned by the gasometer wasn't enough, so I dumbly increased it a little bit. I didn't take the time to investigate why the gasometer was returning an insufficient value.

Reply:
Yes, it's an issue we had since back in 2020. I think there is some broken logic on the CalculateGas method

But this is not a big problem since this is the max gas of a tx so it doesn't increase the cost of the tx

Reply:
> Yes, it's an issue we had since back in 2020. 

Oh right, thx @lubtd for the clarification, so indeed I'm not the one who added that line, what I did instead was to increase the value from 1000 to 2000 (commit 781e90a5e43aa83d3dbe9179eba107cd140711c9).

> I think there is some broken logic on the CalculateGas method

That's weird since the gasometer is only calling `tx.CalculateGas` from the sdk. Unless there's a bug in the sdk, maybe the `txFactory` we pass as a parameter is not complete.